lib/pipeline.js

```diff
@@ -499,6 +487,50 @@ class PipelineModel extends BaseModel {
         return admin;
     }
 
+    /**
+     * This function deletes admins who does not have proper
+     * GitHub permission, and returns an proper admin.
+     * @method getFirstAdmin
+     * @return {Promise}
+     */
+    async getFirstAdmin() {
+        /* eslint-disable no-restricted-syntax */
+        for (const username of Object.keys(this.admins)) {
+            try {
+                /* eslint-disable-next-line no-await-in-loop */
+                return await new Promise((resolve, reject) => {
+                    // Lazy load factory dependency to prevent circular dependency issues
+                    // https://nodejs.org/api/modules.html#modules_cycles
+                    /* eslint-disable global-require */
+                    const UserFactory = require('./userFactory');
+                    /* eslint-enable global-require */
+                    const factory = UserFactory.getInstance();
+
+                    const user = factory.get({
```

Comment:
would this need `await`?

Reply:
`UserFactory.get` is [here](https://github.com/screwdriver-cd/models/blob/master/lib/userFactory.js#L76).  It uses `Promise` only `return` context, so I think `return` will be called after `Promise` is resolved. Thus `factory.get` synchronized  without `await`.

Reply:
I'm sorry, I was wrong. It should be add `await` here. Thank you for comment.